src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -0,0 +1,204 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionBindingOperation.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using EnvDTE;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// Solution level binding by delegating some of the work to <see cref="ProjectBindingOperation"/>
+    /// </summary>
+    internal class SolutionBindingOperation : IBindingOperation, ISolutionRuleStore

```

Comment:
ISolutionRuleStore could be implemented by something else (i.e. the workflow even, or a some state class but in that case will need more API, more tests etc.). It's logically separate from the binding operation and in fact this is the state of the solution binding operation.
It's separate and separable but that would create unnecessary complexity in the product code - we would need an API to write the solution rule sets (when solution binding is preparing) and would also mean that the test code would be somewhat more complex, both of which I wanted to avoid.  The trade-off was injecting those two separable interfaces into the workflow as part of test execution, when in fact the product implements it by the same object.